pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -394,3 +409,38 @@ def _map_usage(response: chat.ChatCompletion | ChatCompletionChunk) -> usage.Usa
             total_tokens=response_usage.total_tokens,
             details=details,
         )
+
+
+async def _map_user_prompt(part: UserPromptPart) -> chat.ChatCompletionUserMessageParam:
```

Comment:
I think better if this (and analogous functions for other models) are methods rather than module-level functions to make it easier to override for users to address issues faster than we make a new release by overriding, if they need to. I'm fine with it being private, just making it a method makes it a lot less code to fix things if you are willing to get your hands dirty with private stuff.